src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -406,10 +406,11 @@ async Task ExecuteInvocation()
 
             try
             {
-                await foreach (var streamItem in enumerable)
+                var enumerator = enumerable.GetAsyncEnumerator(streamCts.Token);
```

Comment:
Should the `enumerator` be disposed?

Reply:
Leave the await foreach and call .WithCancellation instead:

```C#
await foreach (var streamItem in enumerable.WithCancellation(streamCts.Token))
{
}
```

Reply:
Yes it should